agent/structs/structs.go

```diff
@@ -98,6 +99,10 @@ const (
 	// MetaSegmentKey is the node metadata key used to store the node's network segment
 	MetaSegmentKey = "consul-network-segment"
 
+	// MetaWANFederationKey is the mesh gateway metadata key that indicates a
+	// mesh gateway is usable for wan federation.
+	MetaWANFederationKey = "wan-federation"
```

Comment:
Every other reserved meta key I think has the prefix `consul-` since that is our documented reserved prefix. I suggest we prefix this too just for consistency.

Reply:
I wanted to do this, but it trips some validation filter that rejects the registration because the prefix is reserved. Presumably this is because we expect the metadata to be added internally to a consul instance rather than coming in from a normal registration. I couldn't figure out an easy way around that but am totally open to suggestions or pointers.

Reply:
Pushed a commit that refactored how the metadata validation is done so that `consul-wan-federation` is allowed _just_ for service registrations of kind=mesh-gateway.